library/lua/custom-raw-tokens.lua

```diff
@@ -0,0 +1,372 @@
+--[[
+custom-raw-tokens
+Allows for reading custom tokens added to raws by mods
+by Tachytaenius (wolfboyft)
+
+Yes, non-vanilla raw tokens do quietly print errors into the error log but the error log gets filled with garbage anyway
+
+NOTE: This treats plant growths similarly to creature castes but there is no way to deselect a growth, so don't put a token you want to apply to a whole plant after any growth definitions
+]]
+
+local _ENV = mkmodule("custom-raw-tokens")
+
+local customRawTokensCache = {}
+dfhack.onStateChange.customRawTokens = function(code)
+    if code == SC_WORLD_UNLOADED then
+        customRawTokensCache = {}
+    end
+end
+
+local function ensureTable(tableToHoldIn, key)
```

Comment:
I had similar code in quickfort. See #2118

Reply:
#2118 is now merged -- you can use the new global `ensure_key(tableToHoldIn, key)` here